lib/namma-dsl/src/NammaDSL/Generator/Haskell/ApiTree/App.hs

```diff
@@ -0,0 +1,77 @@
+module NammaDSL.Generator.Haskell.ApiTree.App (generateAPITree) where
+
+import Control.Monad (when)
+import Control.Monad.Reader (ask)
+import Data.Functor ((<&>))
+import qualified Data.List.NonEmpty as NE
+import Data.Maybe (fromMaybe)
+import NammaDSL.Config (ApiKind (..), DefaultImports (..))
+import NammaDSL.DSL.Syntax.API
+import NammaDSL.GeneratorCore
+import NammaDSL.Lib hiding (Q, Writer)
+import qualified NammaDSL.Lib.TH as TH
+import qualified NammaDSL.Lib.Types as TH
+import NammaDSL.Utils
+import Prelude
+
+type Writer w = TH.Writer ApiTree w
+
+-- type Q w = TH.Q ApiTree w
+
+generateAPITree :: DefaultImports -> ApiRead -> ApiTree -> Code
+generateAPITree (DefaultImports qualifiedImp simpleImp _packageImports _) apiRead input = generateCode generatorInput
+  where
+    apiServantModulePrefix = apiServantImportPrefix apiRead ++ "."
+
+    generatorInput :: GeneratorInput
+    generatorInput =
+      GeneratorInput
+        { _ghcOptions = ["-Wno-unused-imports"],
+          _extensions = [],
+          _moduleNm = apiServantImportPrefix apiRead,
+          _moduleExports = Just ["API", "handler"],
+          _simpleImports = simpleImp,
+          _qualifiedImports = removeUnusedQualifiedImports codeBody' allQualifiedImports,
+          _packageImports,
+          _codeBody = codeBody'
+        }
+    codeBody' = generateCodeBody (mkCodeBody apiRead) input
+
+    allQualifiedImports :: [String]
+    allQualifiedImports =
+      ((apiServantModulePrefix <>) <$> specModules input)
```

Comment:
We can add **nub** here to reject same imports from dhall and module

Reply:
don't think any duplication is possible here, but anyway fixed